src/dependencies/useStoreDependency.ts

```diff
@@ -83,9 +83,9 @@ function useStoreDependency<Props, DepType>(
 ): DepType {
   enforceDispatcher(dispatcher);
 
-  const [dependencyValue, setDependencyValue] = useState({
+  const [dependencyValue, setDependencyValue] = useState(() => ({
```

Comment:
This is 💯  !

The only other tweak we could consider is literally moving the `const newValue = { dependency: ... }` up above here, then reusing `newValue` both as the initial state and where it is already used. That has the added benefit of skipping an extra eval of the deref in the opposite case (where we _are_ initializing state for the first time)